packages/git/src/browser/git-view-contribution.ts

```diff
@@ -449,6 +742,44 @@ export class GitViewContribution extends AbstractViewContribution<GitWidget>
             tooltip: 'Publish Changes'
         };
     }
+
+    registerScmTitleCommands(registry: ScmTitleCommandRegistry): void {
+        registry.registerCommand(GIT_COMMANDS.REFRESH.id);
+        registry.registerCommand(GIT_COMMANDS.COMMIT_ADD_SIGN_OFF.id);
+    }
+
+    registerScmResourceCommands(registry: ScmResourceCommandRegistry): void {
+    }
+
+    registerScmGroupCommands(registry: ScmGroupCommandRegistry): void {
+    }
+
+    protected async doSignOff() {
+        const { selectedRepository } = this.repositoryProvider;
+        if (selectedRepository) {
+            const [username, email] = await this.getUserConfig(selectedRepository);
+            const signOff = `\n\nSigned-off-by: ${username} <${email}>`;
+            const commitTextArea = document.getElementById(ScmWidget.Styles.INPUT_MESSAGE) as HTMLTextAreaElement;
+            if (commitTextArea) {
+                const content = commitTextArea.value;
+                if (content.endsWith(signOff)) {
+                    commitTextArea.value = content.substr(0, content.length - signOff.length);
+                } else {
+                    commitTextArea.value = `${content}${signOff}`;
+                }
+                this.scmWidget.resize(commitTextArea);
+                commitTextArea.focus();
+            }
+        }
+    }
+
+    protected async getUserConfig(repository: Repository): Promise<[string, string]> {
```

Comment:
This won't work

https://github.com/theia-ide/theia/blob/84284959a3c579f874831b5450e5858670aeee07/packages/git/src/browser/git-widget.tsx#L538-L544

Reply:
used original source